### ISSUE TITLE
possible channel switching issue

### ISSUE THREAD

**Issue by 9000h** [rofafor/vdr-plugin-satip#61]
I do have reduced the test to 2 channels, 1 good and a second bad (RTL7) channel.
If I switch back and for multiple times I end up with no picture on the good channel.
```
Das Erste HD;ARD:11493:HC23M5O35P0S1:S19.2E:22000:5101=27:5102=deu@3,5103=mis@3;5106=deu@106:5104;5105=deu:0:10301:1:1019:0
RTL7;CANALDIGITAAL:12343:hC34S0:S19.2E:27500:518+8190=2:90=dut@4:38:1817,100:2006:53:1097:0
````
[vdr.log.gz](https://github.com/rofafor/vdr-plugin-satip/files/4149686/vdr.log.gz)

this seem to be the point where it's going wrong, vdr satip did only request the pids for the good channel but there is no tune.

**Comment by 9000h** [comment]
did the check in case of a retune make sense here?
https://github.com/rofafor/vdr-plugin-satip/blob/826dea4ba84d8054a5302879c147c55dcb808d81/tuner.c#L224-L227

**Comment by 9000h** [comment]
there is another issue in tsSet the UpdatePids is not needed here as it's done later on in tsLocked
https://github.com/rofafor/vdr-plugin-satip/blob/826dea4ba84d8054a5302879c147c55dcb808d81/tuner.c#L126
https://github.com/rofafor/vdr-plugin-satip/blob/826dea4ba84d8054a5302879c147c55dcb808d81/tuner.c#L156

**Comment by 9000h** [comment]
I see one of the issues come from the latest commit, so may @REELcoder can explain how this makes sense in case of re-tune as it did not work here with as single server dvb-s2/c/t2.
https://github.com/rofafor/vdr-plugin-satip/blob/826dea4ba84d8054a5302879c147c55dcb808d81/tuner.c#L224-L227

**Comment by REELcoder** [comment]
The idea is, that no re-tuning is needed, if you switch from to channel on the same transponder/frequency. But the two channels you mention, are not on the same transponder.

Please activate more satip loging.

**Comment by 9000h** [comment]
if I comment this part out all is fine, as it's happen with minisatip and digibit r1 it should be also happen on a octonet. Can you please try as written in the first post.

**Comment by REELcoder** [comment]
Same effect here. Only with fast switches. If you wait on the 20s tuning timeout on the invalid channel, the valid will be tuned. 

Seems to be a limitation on the ::Action() state machine. Can have a lock into that. Does it happen only with invalid channels?

**Comment by 9000h** [comment]
I do only notice this with invalid channels, and with multiple sat pos. and VDR you have every time some in the channels.conf. I may heart also the background eit/epg scan but the logs are grow so fast with a higher level.
But as I wrote the code is may not needed and I see no drawback without.

**Comment by REELcoder** [comment]
It was intended to prevent from unnecessary retuning if you switch between channels on the same transponder. But I see no problem to remove this in the meantime. I will check what is missing that it works with invalid channel as well.

**Comment by REELcoder** [comment]
My commit 8d84938be3d594aa2d7b60250aaf1c2c71a07335 should fix this. 

The problem was, that that lastParamM was not updated when you switch to crypted invalid channel.

**Comment by 9000h** [comment]
thanks, still testing but looks like without it's still better

**Comment by REELcoder** [comment]
Does ist work? If you see problems, please send a TRAC 0x0009 log.

**Comment by 9000h** [comment]
unfortunately not, on the last switch to chan 1 there is no picture
 
[syslog.txt](https://github.com/rofafor/vdr-plugin-satip/files/4227640/syslog.txt)

**Comment by REELcoder** [comment]
Strange. It is hanging in tsTunned at 22:45:34 when you switch the channel. In tsTunned hasLockM should be false and the "Identical parameters" branch should not get hit in Connect(). But it's happening in your case. Cann you try to add more debug code in Connect() to find out what is happening with hasLockM

**Comment by 9000h** [comment]
it a matter of time here, if you switch with in vdr with key 0 you should be able to reproduce it.

**Comment by 9000h** [comment]
I like to bring this up again as with vdr-2.5.1 and git from today the issue is still happen here
I do have to comment out the "return true" to fix it.
https://github.com/rofafor/vdr-plugin-satip/blob/43b60b1566878104240cbae5b1e322bbacd380e1/tuner.c#L228

**Comment by rofafor** [comment]
Could you try the latest master commit whether it helps?

**Comment by 9000h** [comment]
will do

**Comment by 9000h** [comment]
looks like if frontend reuse is disabled it works as expected, but not with frontend reuse enabled (also not with comment the return mentioned above)

**Comment by 9000h** [comment]
need to correct me
looks like if frontend reuse is disabled it works as expected when the "return" is commented, with frontend reuse the issue is still there

**Comment by rofafor** [comment]
When looking at your latest log, there's no `Identical parameters` within the last zap, so commenting the `return true;` shouldn't have any effect as that code block isn't even executed. IMO, channel switches and retuning goes as it should. There's one `identical parameters` message when properly re-tuning the channel 1 at 21:11:25.

Please, run the satip plugin with one device only and remove all additional plugins like hbbtv that affects devices. Mark into logs when channel isn't visible although it should. Also capture (e.g. tcpdump) port 544 traffic from the same time duration. Running the plugin with one device only makes debugging much easier.
